semgrep-core/bin/Main.ml

```diff
@@ -136,6 +136,17 @@ let supported_langs: string = String.concat ", " keys
 let ncores = ref 1
 (*e: constant [[Main_semgrep_core.ncores]] *)
 
+(* TODO: we may need to put that in AST_generic.ml at some point
+ * and people should bump this number each time they make a modification
+ * to the generic AST (or to some analysis such as Naming_AST.ml which
+ * will have an impact on the marshalled AST). Otherwise, we may
+ * get some segmentation fault as OCaml marshalling is not entirely
+ * type-safe!!
+ *)
+let ast_version = 2
```

Comment:
I suggest that this be the full version (including git commit) of `pfff`, so that any rev to `pfff` invalidates the cache.

Reply:
For development, forcing invalidation whenever `pfff` is dirty would also prevent developer pain.

Reply:
Sgtm. I think brendon generate a version.ml that we can use here